### DISCUSSION TITLE
optional extras

### DISCUSSION THREAD

**Discussion by zalmane** [pyinstaller/pyinstaller#7109] [Features & Development]
I would like to provide my users with a base package and then optional extras. For example, packaging of a set of database drivers.
Is there a mechanism to create an archive with additional wheel files that can be added to an existing one-file bundle or a folder? 
Are there any built in tools to achieve this?

**Comment by bwoodsend** [comment]
PyInstaller isn't going to do this since a PyInstaller built application is supposed to be immutable. You can hack in plugin code by adding directories to `sys.path` at runtime but that's a task better suited to pip rather than adhoc code and to use pip, you need a proper Python environment rather than the bundle PyInstaller produces.

**Comment by zalmane** [comment]
is there a way to create a second pyinstaller bundle and allow the user to
post install these? is there any mechanism i can use to do that?
I am not looking to dynamically load modules, just a way to post install
additional procreate bundles.

**Reply by bwoodsend** [reply]
no
